### ISSUE TITLE
Add `ignore-signatures` option to `[SIMILARITIES]` section

### ISSUE THREAD

**Issue by jnsnow** [PyCQA/pylint#3619]
### Is your feature request related to a problem?
Please describe A clear and concise description of what the problem is.

When comparing code that uses multiple lines for function signatures, it's often the case that classes that implement or extend an interface will have several repeated lines for that interface signature.

e.g. consider a class implementing the `__exit__` interface for a context manager:

```python
class ArbitraryExample:
    def __exit__(self,
                 exc_type: Optional[Type[BaseException]],
                 exc_val: Optional[BaseException],
                 exc_tb: Optional[TracebackType]) -> None:
        pass
```

We're at four lines. If two modules in the project use a context manager, it will trigger the similarity checker.

### Describe the solution you'd like

Implement a 'ignore-signature' option for the [SIMILARITIES] section, and have it default to *yes*.

(In same cases I imagine it might be a legitimate concern that you have many functions with extremely similar interfaces. In the case of special dunder methods at least, we are generally targeting an interface and have no control over the factoring. An ignore option like this might be best, since it seems hard to accurately determine which cases of repetition are fixable and which are not.)

**Comment by adityagupta1089** [comment]
Let me know if I can help here.

**Comment by Pierre-Sassoulas** [comment]
Hey @adityagupta1089 , no one is assigned to the issue so if you want to handle it I can assign it to you :)

**Comment by adityagupta1089** [comment]
@Pierre-Sassoulas cool, I'll probably need some background or otherwise I can go through the code. Do you have some proposal for a solution or should I propose one?

**Comment by Pierre-Sassoulas** [comment]
Nice, I assigned you to the issue :smile: There is already an option for the similarity checker for imports, this one will probably be pretty close to it. You could check that here: https://github.com/PyCQA/pylint/blob/master/pylint/checkers/similar.py#L53 .

**Comment by adityagupta1089** [comment]
@Pierre-Sassoulas I think the PR was merged but did not trigger closure of issue. (I mentioned closes #3619 in description)

**Comment by Pierre-Sassoulas** [comment]
Thanks for notifying me, cleaning these periodically after the fact is a full day of work :D

**Comment by jnsnow** [comment]
For my own record, this was addressed in MR #4474
